csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -70,6 +70,15 @@ public Builder Metadata(string key, string value)
                 return this;
             }
 
+            public Builder Metadata(IEnumerable<KeyValuePair<string, string>> dict)
+            {
+                foreach (KeyValuePair<string, string> entry in dict)
+                {
+                    this.Metadata(entry.Key, entry.Value);
```

Comment:
I'm not sure on the coding styles used in this repo, but on our team we prefer to not use `this` when it isn't necessary.

See (4) here: [https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md)